packages/codemod/src/upgrade-legacy.ts

```diff
@@ -1198,43 +1198,30 @@ const upgradeLegacy = async () => {
 
   steps.push({
     name: "check for usages of invokeWithMiddleware",
-    action: async (stepIndex) => {
-      let errors = 0
-
-      getAllFiles(appDir, [], [], [".ts", ".tsx", ".js", ".jsx"]).forEach((file) => {
-        const program = getCollectionFromSource(file)
-        const invokeWithMiddlewarePath = findCallExpression(program, "invokeWithMiddleware")
-        if (invokeWithMiddlewarePath?.length) {
-          log.error(`\n invokeWithMiddleware found at ${file}. \n`)
-          errors++
-        }
-      })
-
+    action: async () => {
       getAllFiles(path.resolve("pages"), [], [], [".ts", ".tsx", ".js", ".jsx"]).forEach((file) => {
         const program = getCollectionFromSource(file)
-        const invokeWithMiddlewarePath = findCallExpression(program, "invokeWithMiddleware")
-        if (invokeWithMiddlewarePath?.length) {
-          log.error(`\n invokeWithMiddleware found at ${file}. \n`)
-          errors++
+        try {
+          const invokeWithMiddlewarePath = findCallExpression(program, "invokeWithMiddleware")
+          if (invokeWithMiddlewarePath?.length) {
+            invokeWithMiddlewarePath.forEach((path) => {
+              const resolverExpression = j.callExpression(
+                //@ts-ignore
```

Comment:
Could we avoid using `@ts-ignore`?

Reply:
Hey @beerose the reason for the //@ts-ignore is that the types provided for the arguments is only 
![image](https://user-images.githubusercontent.com/83594610/186480722-dec18a81-5628-486f-b375-fcea0c5f0c4f.png)
while the return value when logging the value provides other information like the name that is required for replacing as the function
![image](https://user-images.githubusercontent.com/83594610/186480938-d84d8901-2361-4d1b-b891-c92a64ae1288.png)
So I have now removed the explicit //@ts-ignore by converting the type to of the ```resolverName``` variable to any to avoid type errors

Reply:
@beerose okay I shall try to fix the type mismatch

Reply:
@beerose I have fixed the type error by assigning the type of the returned element, i.e Identifier
https://github.com/siddhsuresh/blitz/blob/a047f589c1f8db5faae0f081fc9d68bda5831d3a/packages/codemod/src/upgrade-legacy.ts#L1209

Reply:
Hey, @dillonraphael I have removed the assertion, instead, I am checking whether the first argument is an Identifier, If so updating. Otherwise, throwing an error with the line number of the failing invokeWithMiddleware call.

Reply:
@dillonraphael Is this change fine?